app/code/Magento/CatalogGraphQl/Test/Integration/Model/ProductsTest.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\CatalogGraphQl\Test\Integration\Model;
+
+use Magento\Framework\App\Request\Http as HttpRequest;
+use Magento\Framework\Serialize\SerializerInterface;
+use Magento\GraphQl\Controller\GraphQl as GraphQlController;
+use Magento\TestFramework\ObjectManager;
+use PHPUnit\Framework\TestCase;
+
+class ProductsTest extends TestCase
+{
+    /**
+     * @var SerializerInterface
+     */
+    private $jsonSerializer;
+    /**
+     * @var ObjectManager
+     */
+    private $objectManager;
+    /**
+     * @var GraphQlController
+     */
+    private $graphql;
+
+    protected function setUp() : void
+    {
+        $this->objectManager = \Magento\TestFramework\Helper\Bootstrap::getObjectManager();
+        $this->graphql = $this->objectManager->get(\Magento\GraphQl\Controller\GraphQl::class);
+        $this->jsonSerializer = $this->objectManager->get(SerializerInterface::class);
+    }
+
+    /**
+     * @magentoAppArea graphql
+     * @magentoDataFixture Magento/CatalogUrlRewrite/_files/product_with_category.php
+     * @magentoDbIsolation disabled
+     * @magentoConfigFixture default_store catalog/seo/product_url_suffix .html
+     */
+    public function testResponseContainsCanonicalURLs(): void
+    {
+        $fixtureSku = 'p002';
+        $query = <<<QUERY
+ {
+   products(filter: {sku: {eq: "$fixtureSku"}})
```

Comment:
Please convert this test into web API functional test.

See [\Magento\GraphQl\Catalog\ProductViewTest::testQueryAllFieldsSimpleProduct](https://github.com/magento/magento2/blob/b3372064f574fd5aece0617009ee317db589febe/dev/tests/api-functional/testsuite/Magento/GraphQl/Catalog/ProductViewTest.php#L35) as an example.